File: 404. Sum of Left Leaves.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sumOfLeftLeaves(self, root: Optional[TreeNode]) -> int:
        return self.helper(root, False)
    def helper(self, root, left):
        if not root:
            return 0
        if not root.left and not root.right:
            if left:
                return root.val
            else:
                return 0
        return self.helper(root.left, True) + self.helper(root.right, False)

File: 404. Sum of Left Leaves.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func sumOfLeftLeaves(root *TreeNode) int {
	return helper(root, false)
}
func helper(root *TreeNode, left bool) int {
	if root == nil {
		return 0
	}
	if root.Left == nil && root.Right == nil {
		if left {
			return root.Val
		} else {
			return 0
		}
	}
	return helper(root.Left, true) + helper(root.Right, false)
}